-f doesn't support new lines as separator

In case one only wants to run the check against a list of files, e.g. changed files in a PR, it'd be useful to accept filenames separated by newlines.
`-f "$files"` [won't work](https://github.com/SaschaMann/v3/runs/1447108037) and `-f $files` will break when there are files with spaces in their name.

```
Unhandled exception. System.IO.DirectoryNotFoundException: Could not find a part of the path '/home/runner/work/v3/v3/languages/julia/concepts/multiple-dispatch/about.md
languages/julia/exercises/concept/fibonacci-iterator/.docs/hints.md
languages/julia/exercises/concept/fibonacci-iterator/.docs/instructions.md
languages/julia/exercises/concept/vehicle-purchase/.docs/instructions.md'.
   at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirectory, Func`2 errorRewriter)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode)
   at System.IO.FileStream.OpenHandle(FileMode mode, FileShare share, FileOptions options)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   at System.IO.StreamReader.ValidateArgsAndOpenPath(String path, Encoding encoding, Int32 bufferSize)
   at System.IO.StreamReader..ctor(String path, Encoding encoding, Boolean detectEncodingFromByteOrderMarks)
   at System.IO.File.InternalReadAllText(String path, Encoding encoding)
   at System.IO.File.ReadAllText(String path)
   at MarkdownLinkChecker.Parser.parseLinks@64.Invoke(Unit unitVar) in /home/runner/work/MarkdownLinkChecker/MarkdownLinkChecker/src/MarkdownLinkChecker/Parser.fs:line 64
   at Microsoft.FSharp.Control.AsyncPrimitives.CallThenInvoke[T,TResult](AsyncActivation`1 ctxt, TResult result1, FSharpFunc`2 part2) in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 386
   at Microsoft.FSharp.Control.Trampoline.Execute(FSharpFunc`2 firstAction) in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 104
--- End of stack trace from previous location ---
   at Microsoft.FSharp.Control.AsyncResult`1.Commit() in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 337
   at Microsoft.FSharp.Control.AsyncPrimitives.RunSynchronouslyInCurrentThread[a](CancellationToken cancellationToken, FSharpAsync`1 computation) in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 870
   at Microsoft.FSharp.Control.AsyncPrimitives.RunSynchronously[T](CancellationToken cancellationToken, FSharpAsync`1 computation, FSharpOption`1 timeout) in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 878
   at Microsoft.FSharp.Control.FSharpAsync.RunSynchronously[T](FSharpAsync`1 computation, FSharpOption`1 timeout, FSharpOption`1 cancellationToken) in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 1142
   at MarkdownLinkChecker.Program.main(String[] argv) in /home/runner/work/MarkdownLinkChecker/MarkdownLinkChecker/src/MarkdownLinkChecker/Program.fs:line 16
/home/runner/work/_temp/8d51e0fb-6063-48dd-b1d5-2788eb61ba6d.sh: line 14:  2683 Aborted                 (core dumped) markdownlinkchecker -v detailed -f "$files"
Error: Process completed with exit code 134.
```

--- a/src/MarkdownLinkChecker.IntegrationTests/OptionsTests.fs
+++ b/src/MarkdownLinkChecker.IntegrationTests/OptionsTests.fs
@@ -63,11 +63,23 @@ module ExcludeOptionTests =
     [<Theory>]
     [<InlineData("-e")>]
     [<InlineData("--exclude")>]
-    let ``Exclude option`` (excludeArg) =
+    let ``Excludes passed using space as separator`` (excludeArg) =
         let results =
             run [| excludeArg
                    "invalid-url-link.md"
                    "invalid-file-link.md"
+                   "--directory"
+                   "Fixtures" </> "ValidAndInvalid" |]
+
+        Assert.ExitedWithoutError(results)
+
+    [<Theory>]
+    [<InlineData("-e")>]
+    [<InlineData("--exclude")>]
+    let ``Excludes passed using newline as separator`` (excludeArg) =
+        let results =
+            run [| excludeArg
+                   "invalid-url-link.md\ninvalid-file-link.md"
                    "--directory"
                    "Fixtures" </> "ValidAndInvalid" |]
 
@@ -78,7 +90,7 @@ module FilesOptionTests =
     [<Theory>]
     [<InlineData("-f")>]
     [<InlineData("--files")>]
-    let ``Files option`` (filesArg) =
+    let ``Files passed using space as separator`` (filesArg) =
         let results =
             run [| filesArg
                    "valid-url-link.md"
@@ -87,6 +99,18 @@ module FilesOptionTests =
                    "Fixtures" </> "ValidAndInvalid" |]
 
         Assert.ExitedWithError(results)
+
+    [<Theory>]
+    [<InlineData("-f")>]
+    [<InlineData("--files")>]
+    let ``Files passed using newline as separator`` (filesArg) =
+        let results =
+            run [| filesArg
+                   "valid-url-link.md\nvalid-file-link.md"
+                   "--directory"
+                   "Fixtures" </> "ValidAndInvalid" |]
+
+        Assert.ExitedWithoutError(results)
 
 module VerbosityOptionTests =
 

--- a/src/MarkdownLinkChecker/Checker.fs
+++ b/src/MarkdownLinkChecker/Checker.fs
@@ -63,7 +63,7 @@ let private checkLinkStatus (link: Link) =
             match link with
             | UrlLink (_) -> checkUrlStatus (linkKey link) |> Async.Catch
             | FileLink (_) -> checkFileStatus (linkKey link) |> Async.Catch
-   
+
         match result with
         | Choice1Of2 status -> return (link, status)
         | Choice2Of2 _ -> return (link, Error)
@@ -119,12 +119,9 @@ let private logCheckedDocument (options: Options) (checkedDocument: CheckedDocum
         let reference = linkReference checkedLink.Link
 
         match checkedLink.Status with
-        | Found ->
-            options.Logger.Detailed(sprintf "✅ (%d,%d): %s" position.Line position.Column reference)
-        | NotFound ->
-            options.Logger.Minimal(sprintf "❌ (%d,%d): %s" position.Line position.Column reference)
-        | Error ->
-            options.Logger.Minimal(sprintf "❌ (%d,%d): %s (error)" position.Line position.Column reference)
+        | Found -> options.Logger.Detailed(sprintf "✅ (%d,%d): %s" position.Line position.Column reference)
+        | NotFound -> options.Logger.Minimal(sprintf "❌ (%d,%d): %s" position.Line position.Column reference)
+        | Error -> options.Logger.Minimal(sprintf "❌ (%d,%d): %s (error)" position.Line position.Column reference)
 
 let private checkDocument (options: Options) (checkedLinks: Map<string, LinkStatus>) (document: Document) =
     let checkedLinks = toCheckedLinks checkedLinks document

--- a/src/MarkdownLinkChecker/Options.fs
+++ b/src/MarkdownLinkChecker/Options.fs
@@ -1,5 +1,6 @@
 module MarkdownLinkChecker.Options
 
+open System
 open MarkdownLinkChecker.Logging
 
 open CommandLine
@@ -70,9 +71,14 @@ let private parseMode (mode: string) =
     | "urls" -> CheckUrlLinks
     | _ -> CheckAllLinks
 
+let private parseFiles (files: string seq): string [] =
+    files
+    |> Seq.collect (fun file -> file.Split('\n', StringSplitOptions.RemoveEmptyEntries))
+    |> Seq.toArray
+
 let private fromCommandLineOptions (options: CommandLineOptions) =
-    { Files = Array.ofSeq options.Files
-      Exclude = Array.ofSeq options.Exclude
+    { Files = parseFiles options.Files
+      Exclude = parseFiles options.Exclude
       Directory =
           options.Directory
           |> Option.defaultWith System.IO.Directory.GetCurrentDirectory

